src/main/java/org/jboss/naming/remote/client/InitialContextFactory.java

```diff
@@ -61,6 +70,71 @@
      * @throws NamingException if an error occurs
      */
     public Context getInitialContext(final Hashtable<?, ?> environment) throws NamingException {
-        return delegate.getInitialContext(environment);
+        return delegate.getInitialContext(findAndCreateClientProperties(environment));
+    }
+
+    private static ClassLoader getClientClassLoader() {
+        final ClassLoader tccl = secureGetContextClassLoader();
+        if (tccl != null) {
+            return tccl;
+        }
+        return InitialContextFactory.class.getClassLoader();
+    }
+
+    private Properties findAndCreateClientProperties(final Hashtable<?, ?> env) {
+        // First load the props file if it exists
+        Properties props = findClientProperties();
```

Comment:
Since you're creating a `Properties` either way, you could create it and pass it in to this method which would populate it.